mirror_sparse_matrix/Cargo.toml

```diff
@@ -1,15 +1,15 @@
 [package]
 name = "mirror_sparse_matrix"
-version = "0.1.14"
+version = "0.1.15"
 authors = ["David Jaffe <david.jaffe@10xgenomics.com>"]
 license = "MIT"
 description = "Some tools that are 'internal' for now because they are insufficiently refined and unstable, but which are used by other 'public' crates."
 edition = "2018"
 repository = "https://github.com/10XGenomics/rust-toolbox"
 
 [dependencies]
-binary_vec_io = "0.1"
+binary_vec_io = { version = "0.2", path = "../binary_vec_io" }
```

Comment:
Was this actually a breaking change for `mirror_sparse_matrix`?  If not then it would be better to widen the version range, to avoid a horrid mess trying to keep dependencies from being duplicated as dependabot updates crates one by one.  Similarly for other updates.

Reply:
Not really, since all errors are `.unwrap()` in `mirror_sparse_matrix`.

For `pretty_trace` and `binary_vec_io` it is more complicated. Changing the `Error` type would be a breaking change in semver (especially because `failure::Error` is not `std::error`-compatible, and will cause breakage in any downstream crate not using `failure`). I think it is better to indicate that with a version bump?

As for widen version range, keeping it to `0.1` here isn't going to make things worse (if the `binary_vec_io` version bump happens)? Won't `cargo` prefer to keep `0.1` (based on `failure`) instead of `0.2` (based on `thiserror`), and potentially add `failure` back to the dependency tree?

Reply:
The issue is if the dependent crate has another dependency on `pretty_trace`, either directly or through some other crate they haven't updated yet, in which case then they end up with two different versions of `pretty_trace` and still have `failure` in their dependency tree.

Reply:
I went with more conservative patch bumps for `pretty_trace` and `binary_vec_io` and left `mirror_sparse_matrix` and `vdj_ann` untouched on latest commit.